ibis/backends/flink/__init__.py

```diff
@@ -318,27 +350,50 @@ def create_table(
 
         if obj is None and schema is None:
             raise exc.IbisError("The schema or obj parameter is required")
+        if obj is not None and schema is not None:
+            raise exc.IbisError(
+                "Parameters `schema` and `obj` cannot be provided together. "
+                "Exactly one of them must be provided."
+            )
 
         if overwrite:
             self.drop_table(name=name, catalog=catalog, database=database, force=True)
 
         if isinstance(obj, pa.Table):
             obj = obj.to_pandas()
+
         if isinstance(obj, pd.DataFrame):
             qualified_name = self._fully_qualified_name(name, database, catalog)
             table = self._table_env.from_pandas(obj)
             # in-memory data is created as views in `pyflink`
             # TODO(chloeh13q): alternatively, we can do CREATE TABLE and then INSERT
             # INTO ... VALUES to keep things consistent
             self._table_env.create_temporary_view(qualified_name, table)
+
         if isinstance(obj, ir.Table):
-            # TODO(chloeh13q): implement CREATE TABLE for expressions
-            raise NotImplementedError
+            # TODO (mehmet): Double check this with the reviewers!
+            if not self.list_tables(name):
+                return ibis.memtable(obj, schema=obj.schema())
+            else:
+                return self.create_view(
+                    name=name,
+                    obj=obj,
+                    database=database,
+                    catalog=catalog,
+                    temporary=temp,
+                    overwrite=overwrite,
+                )
```

Comment:
This is confusing to me.  If a user defines an `UnboundTable` that they would like to create in the Flink backend, they're going to end up with a `view` instead -- I think a call to `create_table` should create a table.

Reply:
Many thanks @gforsyth for all your comments. They urged me to go back and discuss the behavior of `create_table()` for Flink with @deepyaman (CC: @chloeh13q, @zhenzhongxu). Here is our understanding / conclusion:
* Flink needs a `source connector` to create a (regular) table.
    *  Ref: [Create Table](https://nightlies.apache.org/flink/flink-table-store-docs-master/docs/development/create-table/)
* In-memory data does not have a connector information. This is why, we can only create a `temporary view` in Flink with in-memory data. (`Views` are referred to as `virtual tables` in Flink.)
    * Ref: [Create Tables in the Catalog](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/common/#create-tables-in-the-catalog)
    > Tables can be either virtual (VIEWS) or regular (TABLES). VIEWS can be created from an existing Table object, usually the result of a Table API or SQL query. TABLES describe external data, such as a file, database table, or message queue.

Given these, we have two options in `create_table()` for Flink backend:
1. Require the connector information to be provided. This would imply that we do not support creating table from `in-memory` data (e.g., `pd.Dataframe`, `pa.Table`).
2. Create a view for in-memory data, as currently done.
3. As far as we understand it, `ir.Table` does not have connector information. If we go with option-1, then we won't support creating table from `ir.Table`. If we go with option-2 and `obj` points to an in-memory data (e.g., `obj = ibis.memtable([("a", 1.0)], columns=["a", "b"])`), we create a view. If `obj` is an `UnboundTable`, we do not support it.

Can you please share your opinion and feedback on these conclusions?

Reply:
We'll probably have to revisit this after this PR and the work in #7640 are consolidated.

I think option 2 is fine for now.  I think there should be support for unbound tables, but that may require some notion of a "default" connector, which may not make sense -- that's still unclear to me.

Reply:
Thanks @gforsyth. I implemented option 2 and consolidated my changes with those in https://github.com/ibis-project/ibis/pull/7640.